src/Repository/DonationRepository.php

```diff
@@ -18,4 +19,12 @@ public function findOneByUuid(string $uuid): ?Donation
     {
         return $this->findOneByValidUuid($uuid);
     }
+
+    public function findByEmailAddressOrderedByDonationAt(Adherent $adherent, string $order = 'ASC'): array
```

Comment:
The ordered field is `donatedAt`, so you should rename this method in `findByEmailAddressOrderedByDonatedAt`

Reply:
And IMO, you could pass directly the `$adherent->getEmailAddress()` instead of the `Adherent` object :
`public function findByEmailAddressOrderedByDonatedAt(string $emailAddress, string $order = 'ASC')`